### PR TITLE
feat: 999 role switching screen

### LINKED ISSUE #999: Role switching screen
Implement the role switching component under a user's profile

https://www.figma.com/file/n5TRI6CkYd7kwIcOULCP7y/User-role-and-organization-selection?type=design&node-id=77-43450&mode=design&t=uo1vwJZPEXPpyDDJ-4

- [ ] users can see a list of roles
- [ ] users can switch role 
- [ ] user's org is displayed on top right of the header
- [ ] user initial avatar

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Closes #999
[Figma link](https://www.figma.com/file/n5TRI6CkYd7kwIcOULCP7y/User-role-and-organization-selection?type=design&node-id=227-39679&mode=dev)

Allows users to switch organization in the header profile menu

#### New
- UserButton for BCHeader
- New Avatar component that displays user's initial instead of an image

#### Changed
- BCHeader
- MyProfile
- updated OrganizationSelection component so it displays a simple view in profile
- disabled some buttons on header that is not used

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar--frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1016-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1016-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-16-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1016-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1016-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/views/LoginOrgSelection/constants.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const TEXT = {
+  subtitle: "select which organization you're representing."
+};
```
Comment (mgaseta) [inline frontend/src/views/LoginOrgSelection/constants.ts:3]: Does this need to be in a separate file? It seems to be used in only one place...

File: frontend/src/views/LoginOrgSelection/styles.scss [hunk 2]
```diff
@@ -1,8 +1,8 @@
 @use '@carbon/type';
 @use '@bcgov-nr/nr-theme/design-tokens/variables.scss' as vars;
 
-.login-role-selection-grid {
-  background: linear-gradient(0deg, rgba(227, 168, 43, 0.80) 0%, rgba(0, 115, 230, 0.80) 79.97%, rgba(35, 64, 117, 0.80) 100%);
+.login-org-selection-grid {
+  background: linear-gradient(180deg, #E0EFFF 0%, #F2EEDB 100%);
```
Comment (mgaseta) [inline frontend/src/views/LoginOrgSelection/styles.scss:5]: I think I may know the answer here, but just to confirm, these colors aren't in the design system, right? If so, maybe we need to warn the UX team.

**Review by mgaseta — COMMENTED** [review]
Good job! :100: :1st_place_medal: 
Added only a few comments and questions.

File: frontend/src/views/LoginOrgSelection/styles.scss (hunk 2 above)
Comment (craigyu) [reply frontend/src/views/LoginOrgSelection/styles.scss:5]: Yes, i was gonna leave a comment here but forgot. These colors are indeed not in the `colors` config

File: frontend/src/components/Avatar/style.scss [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+.initial-avatar-sm {
+  width: 2rem;
+  height: 2rem;
+  min-width: 2rem;
+  min-height: 2rem;
+  font-size: 0.75rem;
+  font-weight: 700;
+  line-height: 1.125rem;
+  color: var(--#{vars.$bcgov-prefix}-text-text-primary, colors.$gray-100);
```
[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/src/components/Avatar/style.scss:22]: smart

[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/src/components/Avatar/style.scss]: https://github.com/bcgov/nr-spar/pull/1016/commits/631099a690c9faf1d5f3ff8227b00ff846b75610

File: frontend/src/views/LoginOrgSelection/constants.ts (hunk 1 above)
Comment (craigyu) [reply frontend/src/views/LoginOrgSelection/constants.ts:3]: the previous design had a longer string, in this case i'll get rid of it

Comment (craigyu) [reply frontend/src/views/LoginOrgSelection/constants.ts]: https://github.com/bcgov/nr-spar/pull/1016/commits/d5fc7ffb924e5f2cb5475f9954a73b6cb778a4e3

**Review by mgaseta — APPROVED** [review]
LGTM!
Switch time!

**Pull request merged**